data.js

```diff
@@ -198,6 +198,12 @@ function save (callback) {
 
         // for update queries
 
+        if (err) {
+            callback("Update error: " + err);
+            self.emit('saved', err);
```

Comment:
@IonicaBizau I don't see any evident listeners to `saved` (there might be dynamic ones but I doubt it). But everywhere the usage is the same `self.emit('saved', err, data, isInsert)`. So we are good to go.